api/utils_celery.py

```diff
@@ -59,8 +61,20 @@ def get_inactive_task_pages():
 
 
 def process_changed_articles():
-    for language, page_title in iter_changed_pages(logger):
+    for language, page_title, change in iter_changed_pages(logger):
         # print(len(get_inactive_task_pages()))
+        if change.get('type') == 'log' and change.get('log_type') == 'delete' and \
+                change.get('log_action') in ['delete', 'revision']:
+            page_id = get_page_id_from_log_id(page_title, language, change.get('log_id'))
+            if page_id is None:
+                logger.error(
+                    "Failed to retrieve page ID for '" + page_title + "'\n" +
+                    "The associated pickle file must manually be deleted."
```

Comment:
I'm still observing the event stream worker exit whenever it goes down this path.  In this case, it was from a revision being hidden: https://en.wikipedia.org/w/index.php?title=Reese%27s_Puffs&action=history

Reply:
So, I had to restart my computer and I lost my bash history... and now I don't recall how I got EventStreams working! I have RabbitMQ running, as well as the web server, then I run `celery -A wikiwho_api worker --loglevel=INFO` inside the virtual environment. I thought that was it but I must be missing something because I'm not seeing any changes coming through in the log. How do you test on your local?

Anyway, I definitely observed this correctly deleting the pickle file yesterday! So I _think_ you shouldn't be seeing this error. But you're right about exiting early. We shouldn't return in here at all... I've pushed an update to fix that.

Reply:
`python manage.py celery_changed_articles` starts the watcher.

Reply:
In this case, there was no pickle to delete, but it didn't go down the try-to-delete-pickle path because `page_id` must have been `None`. In this case, all I did was turn it on and wait... sadly, there were no speedy candidates that I could test by deleting.

Reply:
> `python manage.py celery_changed_articles` starts the watcher.

That's the one! Thanks. I'll do more testing now, but I believe it should work based on my observations yesterday. Here's what I did, albeit a bit hacky:
* Change the code to look for namespace 2 at [api/handler.py#L229](https://github.com/wikimedia/wikiwho_api/blob/2acbf56aaead66d58e8b917df63d1ef273bf28a9/api/handler.py#L229) and [api/event_streams.py#L70](https://github.com/wikimedia/wikiwho_api/pull/5/files#diff-625b493c9b7ad211505526ff85d2f0ca5d6cdbda04196e9d3a589d5cf490daf9R70)
* Use the WikiWho API to create the pickle for a page in your userspace
* Edit the page and do revdel's and deletion there
* Observe the pickle was deleted

I don't think changing the namespace would matter for testing purposes, but maybe it does... I'm going to be bold and do a live test in the mainspace to be sure. I tried using testwiki but we're apparently using some i18n library that validates language codes, and `test` of course is not a real language. Same applies to Meta and those are the only wikis where I'm an admin.